packages/design-system/src/helpers/handleResponsiveValues.ts

```diff
@@ -40,52 +26,146 @@ type ResponsiveCSSProperties = Pick<
   | 'paddingInline'
   | 'paddingInlineStart'
   | 'paddingInlineEnd'
+  | 'gap'
+>;
+
+type Breakpoint = 'initial' | 'small' | 'medium' | 'large';
+
+type ResponsiveProperty<T> =
+  | {
+      [key in Breakpoint]?: T;
+    }
+  | T;
+
+// shorthand Responsive property ex: padding, marging, etc
+type ShorthandResponsiveProperty = ResponsiveProperty<
+  string | keyof DefaultTheme['spaces'] | Array<string | keyof DefaultTheme['spaces']>
 >;
+// Individual Responsive property ex: padding-top, margin-left, etc
+type IndividualResponsiveProperty = ResponsiveProperty<string | keyof DefaultTheme['spaces']>;
+
+type ResponsiveValue<TCSSProp extends keyof ResponsiveCSSProperties = any> = TCSSProp extends 'padding' | 'margin'
+  ? ShorthandResponsiveProperty
+  : IndividualResponsiveProperty;
+
+type ResponsiveValues<TCSSProp extends keyof ResponsiveCSSProperties = any> = {
+  [K in keyof ResponsiveCSSProperties]?: ResponsiveValue<TCSSProp>;
+};
+
+const mappedLogicalProps = {
+  padding: ['padding-block-start', 'padding-inline-end', 'padding-block-end', 'padding-inline-start'],
+  paddingTop: 'padding-block-start',
+  paddingRight: 'padding-inline-end',
+  paddingBottom: 'padding-block-end',
+  paddingLeft: 'padding-inline-start',
+  margin: ['margin-block-start', 'margin-inline-end', 'margin-block-end', 'margin-inline-start'],
+  marginLeft: 'margin-inline-start',
+  marginRight: 'margin-inline-end',
+  marginTop: 'margin-block-start',
+  marginBottom: 'margin-block-end',
+};
 
-export type ResponsiveValue<TCSSProp extends keyof ResponsiveCSSProperties = any> =
-  | ResponsiveValueObject
-  | DefaultThemeOrCSSProp<'spaces', TCSSProp>
-  | ResponsiveValueTuple;
+const fillCssValues = (value: Array<string | keyof DefaultTheme['spaces']>) => {
```

Comment:
This needs some JSDoc explaining what it does and why it does it etc. It's not clear just reading the code.